packages/react/src/components/select/select.tsx

```diff
@@ -51,37 +55,24 @@ interface ContentProps extends CustomSelectProps {
   borderless?: boolean;
   native?: boolean;
   open?: boolean;
-  onEmptyChange: (empty: boolean) => void;
   onOpenChange: (open: boolean) => void;
 }
 
 function Content({
   borderless,
   native,
   open,
-  onChange,
-  onEmptyChange,
   onOpenChange,
   ...restProps
 }: ContentProps) {
-  if (native)
-    return (
-      <NativeSelect
-        {...restProps}
-        onChange={(event) => {
-          onEmptyChange(!event.currentTarget.value);
-          onChange?.(event);
-        }}
-      />
-    );
+  if (native) return <NativeSelect {...restProps} />;
 
   return (
     <CustomSelect
       {...restProps}
       borderless={borderless}
       open={open}
       onOpenChange={onOpenChange}
-      onSelectOption={(value) => onEmptyChange(!value)}
```

Comment:
This should've been the writing on the wall, needing a custom event rather than `onChange`.
Something to watch out for next time